components/places/src/api/places_api.rs

```diff
@@ -450,32 +413,24 @@ impl PlacesApi {
         Ok(())
     }
 
-    /// Create a new SqlInterruptScope for the syncing code
-    ///
-    /// The syncing code expects a SqlInterruptScope, but it's never been actually hooked up to
-    /// anything.  This method returns something to make the compiler happy, but we should replace
-    /// this with working code as part of #1684.
-    pub fn dummy_sync_interrupt_scope(&self) -> SqlInterruptScope {
-        SqlInterruptScope::new(Arc::new(AtomicUsize::new(0)))
+    // Deprecated method to open a PlacesDb connection using a PlacesApi
+    //
+    // The new method to access the database is through the `PlacesConnection`.  However, there are
+    // a several parts of the code that still use this: autocomplete, the unittests, and the
+    // separated tests. We should refactor this code to use `PlacesConnection`, instead of
+    // `PlacesDb`.  This means either refactoring it to use that `PlacesConnection` API, or by
+    // giving it access to the `with_conn()` method and continuing to use the lower-level
+    // `PlacesDb` API.
+    pub fn open_connection(&self, conn_type: ConnectionType) -> Result<PlacesDb> {
+        PlacesDb::open(&self.db_name, conn_type, self.id, self.coop_tx_lock.clone())
     }
+}
 
-    // Deprecated/Broken interrupt handler method
-    // This should be removed as part of https://github.com/mozilla/application-services/issues/1684
-    //
-    // There are two issues with this one:
-    //   - As soon as this method returns, the sync connection will be dropped, which means the
-    //     SqlInterruptHandle will not work.
-    //   - We want the sync connection to be lazy, but we call this on initialization and force a
-    //     connection to be created.
-    // We have to use Arc in the return type to be able to properly
-    // pass the SqlInterruptHandle as an object through Uniffi
-    pub fn new_sync_conn_interrupt_handle(&self) -> Result<Arc<SqlInterruptHandle>> {
-        // Probably not necessary to lock here, since this should only get
-        // called in startup.
-        let _guard = self.sync_state.lock();
-        let conn = self.get_sync_connection()?;
-        let db = conn.lock();
-        Ok(Arc::new(db.new_interrupt_handle()))
+impl shutdown::ShutdownInterrupt for PlacesApi {
+    fn interrupt(&self) {
+        self.read_connection.interrupt_handle.interrupt();
+        self.write_connection.interrupt_handle.interrupt();
+        // TODO interrupt the sync connection, if it's open
```

Comment:
I got to a point where I realized there was already a ton of changes and I should discuss them with someone before making more.

Reply:
Yeah, I think we should see if we can use a lighter touch here.